notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java

```diff
@@ -1442,13 +1442,13 @@ public Response getFeaturedImageIllustration(@Context Request request,
   }
   
   @POST
-  @Path("/note/view/{noteId}/{lang}")
+  @Path("/note/view/{noteId}")
   @Operation(summary = "mark a note as viewed", method = "POST", description = "This marks a note as viewed.")
   @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Request fulfilled"),
       @ApiResponse(responseCode = "401", description = "User not authorized to get the note"),
       @ApiResponse(responseCode = "500", description = "Internal server error") })
   public Response markNoteAsViewed(@Parameter(description = "News id") @PathParam("noteId") String noteId,
-                                   @Parameter(description = "News target lang") @PathParam("lang") String lang) {
+                                   @Parameter(description = "News target lang") @QueryParam("lang") String lang) {
```

Comment:
What is the reason of this change ?

Reply:
because when lang is null in version language i will have incorrect non matching endpoint ends with ...id/